tvb_contrib/tvb/contrib/cosimulation/models/reduced_wong_wang_exc_io.py

```diff
@@ -0,0 +1,334 @@
+# -*- coding: utf-8 -*-
+#
+#
+#  TheVirtualBrain-Contributors Package. This package holds simulator extensions.
+#  See also http://www.thevirtualbrain.org
+#
+# (c) 2012-2020, Baycrest Centre for Geriatric Care ("Baycrest") and others
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+
+"""
+Models based on Wong-Wang's work.
+.. moduleauthor:: Dionysios Perdikis <dionysios.perdikis@charite.de>
+"""
+
+from numba import guvectorize, float64
+from tvb.simulator.models.wong_wang import ReducedWongWang as TVBReducedWongWang
+from tvb.simulator.models.base import numpy, ModelNumbaDfun
+from tvb.basic.neotraits.api import NArray, Final, List, Range
+
+
+@guvectorize([(float64[:],)*11], '(n),(m)' + ',()'*8 + '->(n)', nopython=True)
+def _numba_update_non_state_variables_before_integration(S, c, a, b, d, w, jn, r, g, io, newS):
+    "Gufunc for reduced Wong-Wang model equations."
+
+    newS[0] = S[0]  # S
+    newS[1] = S[1]  # Rint
+
+    cc = g[0]*jn[0]*c[0]
+    newS[4] = w[0] * jn[0] * S[0] + io[0] + cc  # I
+    if r[0] <= 0.0:
```

Comment:
This and line 56 below are a little hard to understand.  Are these conditionals expressing different numerics or different code paths for different components?  What does `r[0] <= 0.0` mean? Why below branch on `S[1] < 40.0`?

Reply:
The first one is indeed a particular implementation for different code for the nodes that are updated by NEST.
The second one has to do with a very particular use-case.
I moved this whole models folder under tvb.contrib.scripts, because they are too specific to particular use-cases of TVB-multiscale, which are due to change also upon refactoring TVB-multiscale to use this code..